### PR TITLE
Improve user resolve from residence organization to validate the domain qualifier

### PR DESCRIPTION
**Pull request by sadilchamishka**
## Purpose
$subject

### CODE REVIEW

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationUserResidentResolverServiceImpl.java [hunk 1]
```diff
@@ -60,21 +61,39 @@ public Optional<User> resolveUserFromResidentOrganization(String userName, Strin
...
+        String domainAwareUsername = null;
+
         try {
             if (userName == null && userId == null) {
                 throw handleClientException(ERROR_CODE_NO_USERNAME_OR_ID_TO_RESOLVE_USER_FROM_RESIDENT_ORG);
             }
+            /*
+                Separate out domain name and domain aware username.
+             */
+            if (userName != null && userName.split(DOMAIN_SEPARATOR).length > 1) {
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationUserResidentResolverServiceImpl.java:74]: Let's use userName.indexOf(DOMAIN_SEPARATOR) > 0 condition

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationUserResidentResolverServiceImpl.java [hunk 2]
```diff
@@ -60,21 +61,39 @@ public Optional<User> resolveUserFromResidentOrganization(String userName, Strin
...
+                domainName = userName.split(DOMAIN_SEPARATOR)[0];
+                domainAwareUsername = userName.split(DOMAIN_SEPARATOR)[0];
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationUserResidentResolverServiceImpl.java:76]: Is this correct?

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationUserResidentResolverServiceImpl.java (hunk 1 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationUserResidentResolverServiceImpl.java]: Fixed by -  https://github.com/wso2-extensions/identity-organization-management-core/pull/33/commits/e24babcbb2baa7797fee9295d12bb9220d771f47

Used the suggested `UserCoreUtil.extractDomainFromName(userName)` method to extract the domain from the username

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationUserResidentResolverServiceImpl.java (hunk 2 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationUserResidentResolverServiceImpl.java]: Fixed by - https://github.com/wso2-extensions/identity-organization-management-core/commit/e24babcbb2baa7797fee9295d12bb9220d771f47

Removed this logic and get the user store domain via this method `UserCoreUtil.extractDomainFromName(userName)`.

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**